contracts/Oxo.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity 0.4.15;
+contract Oxo{
+
+    address owner;
+    uint8 public playerNum = 1;
+    uint public bet;
+    uint public movesCount = 1;
+    address player1Addr;
+    address player2Addr;
+    bool gameOn = false;
+
+
+    function Oxo() public {
+        owner = msg.sender;
+    }
+
+    mapping (address => uint8) public players;
+
+    uint8[3][3] public field;
+
+    event Move(uint8 col, uint8 row, string occupied);
+    event JoinGame(address indexed addr, uint8 player, uint bet, string str);
+    event Win(address indexed addr, uint8 player, uint gain, string str);
+    event DeadHeat(uint bet, address indexed player1Addr, address indexed player2Addr, string str);
+    event Debug(string str);
+    event GameOver(uint contractBalance);
+
+    function joinGame() public payable {
```

Comment:
What will happen if you make a bet and no one join to you? Are this money lost for ever?

Reply:
thanks for the help, improved code & tests according to comments